### PR TITLE
feat: magento 2 local installation and configuration with vsf - cli tool 

### PR DESCRIPTION
**Pull request by skirianov**
- Created a new CLI tool to install and configure Magento 2 with Vue Storefront on the local machine.
- Implemented `clack/prompts` https://github.com/natemoo-re/clack/tree/main/packages/prompts and migrated all CLI functionality to this new UI
- Fixed previous tests and wrote new tests covering most of the functions

This new tool will be released as **Beta**.

## Description

This will allow you to install and configure Magento 2 locally in under 10 minutes. It should greatly reduce the time from 0 to test the setup locally. Additionally using new CLI UX prompts system by https://github.com/natemoo-re

## Related Issue

No issue created

## Motivation and Context

For the sake of a better DX.

## How Has This Been Tested?

- Fixed previous tests to align with the new UI system
- Wrote new tests to cover most of the functions (except for the Docker ones)
- Tested on Linux and Mac OS, pending tests on Windows

## Screenshots:

https://user-images.githubusercontent.com/74229951/222170924-5c4084f7-c2cf-4ccd-b8c6-e4d3ef5bb102.mp4

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.

#### Changelog
- [ ] I have updated the Changelog ([V1](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md)) [v2](https://docs-next.vuestorefront.io/contributing/creating-changelog.html) and mentioned all breaking changes in the public API.
- [ ] I have documented all new public APIs and made changes to existing docs mentioning the parts I've changed so they're up to date.

#### Tests
- [x] I have written test cases for my code
- [x] I have tested my Pull Request on production build and (to my knowledge) it works without any issues
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

> I tested manually my code, and it works well with both:
- [ ] Default Theme
- [ ] Capybara Theme

#### Code standards
- [x] My code follows the code style of this project.

- [x] I have followed [naming conventions](https://github.com/kettanaito/naming-cheatsheet)

#### Docs
- [x] My change requires a change to the documentation. - @mattmaribojoc 
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/cli/src/domains/magento2/docker/installMagentoImage.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      'curl',
+      [
+        '-s',
+        'https://raw.githubusercontent.com/markshust/docker-magento/master/lib/onelinesetup'
+      ],
+      options
+    );
+    const bash = spawn(
+      'bash',
+      ['-s', '--', magentoDomainName, '2.4.4'],
```
Comment (Igloczek) [inline packages/cli/src/domains/magento2/docker/installMagentoImage.ts:32]: I believe Magento version should be configurable, with a default that is the latest version (so it needs to be automated, or manually updated a few times a year)

But... the underlying API is already handling that, so I guess you can just remove the version from here.

File: packages/cli/__tests__/domains/magento2/confirmOverwrite.spec.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  it('user can confirm overwrite', async () => {
+    const answer = async () => {
+      expect(output).toContain('command.generate_store.magento.overwrite');
+
+      io.send(ENTER_KEY);
+
+      await wait(100);
+
+      expect(output).toContain('🗑️  Deleting the existing directory');
```
Comment (Igloczek) [inline packages/cli/__tests__/domains/magento2/confirmOverwrite.spec.ts:48]: Shouldn't that also come from translations?

**Review by Igloczek — APPROVED** [review]
Note: I didn't review everything, just the new Magento 2 stuff. Haven't tried to run this locally etc.

In general looks fine, just a few minor things in the comments.

File: packages/cli/src/domains/magento2/prompts/confirmOverwrite.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  const sp = spinner();
+
+  if (overwrite) {
+    sp.start(picocolors.cyan('🗑️  Deleting the existing directory'));
+    await fs.rmSync(magentoDirName, { recursive: true, force: true });
+    await fs.mkdirSync(magentoDirName);
+    sp.stop(picocolors.green('🗑️  Directory deleted'));
+  }
+
+  // eslint-disable-next-line max-depth
```
[the remark this answers is not in the document]
Comment (skirianov) [reply packages/cli/src/domains/magento2/prompts/confirmOverwrite.ts:40]: true, copy pasted from the main script and forgot to remove

File: packages/cli/src/domains/magento2/docker/installMagentoImage.ts (hunk 1 above)
Comment (skirianov) [reply packages/cli/src/domains/magento2/docker/installMagentoImage.ts:32]: I'm not so sure about that 
```# Run this automated one-liner from the directory you want to install your project.
curl -s https://raw.githubusercontent.com/markshust/docker-magento/master/lib/onelinesetup | bash -s -- magento.test 2.4.5-p1 community```

https://github.com/markshust/docker-magento - they are using a newer version, but our docs are still mentioning 2.4.4 - can't guarantee that newer works too. 

Need to test

File: packages/cli/__tests__/domains/magento2/confirmOverwrite.spec.ts (hunk 2 above)
Comment (skirianov) [reply packages/cli/__tests__/domains/magento2/confirmOverwrite.spec.ts:48]: should :(

File: packages/cli/__tests__/domains/magento2/confirmOverwrite.spec.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+      await wait(100);
+
+      expect(output).toContain('🗑️  Directory deleted');
+    };
+
+    wait(100).then(answer);
```
[the remark this answers is not in the document]
Comment (skirianov) [reply packages/cli/__tests__/domains/magento2/confirmOverwrite.spec.ts:55]: True! Good catch thanks!

[the remark this answers is not in the document]
Comment (skirianov) [reply packages/cli/__tests__/domains/magento2/confirmOverwrite.spec.ts]: Interesting, this fails with await. confirmOverwrite function is not being called 🤔

File: packages/cli/src/domains/magento2/functions/handleGraphQL.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            if (err) return console.log(err);
+          }
+        );
+      }
+    );
+  };
+
+  return new Promise((resolve, reject) => {
+    const child = spawn(
+      'bin/composer require caravelx/module-graphql-config && bin/magento module:enable Caravel_GraphQlConfig && bin/magento setup:upgrade && bin/magento setup:di:compile && bin/magento setup:static-content:deploy -f',
```
[the remark this answers is not in the document]
Comment (skirianov) [reply packages/cli/src/domains/magento2/functions/handleGraphQL.ts]: We need it actually. Tried without it and it results in the error. So, reverting this change back

**Pull request merged**